cross-cluster/openshift/x-cluster-migrate

```diff
@@ -0,0 +1,392 @@
+      _filters=${_parameterFilters}
+      if [[ ${_paramFile} == *".nsp-source."* ]]; then
+        _filters=${_sourceParameterFilters}
+      elif [[ ${_paramFile} == *".nsp-target."* ]]; then
+        _filters=${_destParameterFilters}
+      fi
+      printStatusMsg ${_filters}
+      cat ${_paramFile} | sed ${_filters} > ${_paramFile}.tmp
+      rm ${_paramFile}
+      mv ${_paramFile}.tmp ${_paramFile}
+    done
+  )
+}
+
+initialize(){
+  echo -e "\nInitializing environment settings ...\n"
+
+  # Write a local settings file ...
+  echo "Writing local settings file (settings.local.sh) ..."
+  echo "export GIT_URI=${GIT_URI}" > settings.local.sh
+  echo "export GIT_REF=${GIT_REF}" >> settings.local.sh
+
+  oc config get-contexts | sed 's/*/ /g' | awk '{print $1}' | sed '1 d'
+  printStatusMsg "Please enter your destination cluster from the list above:"
+  read DST_CONTEXT
+  echo "export DST_CONTEXT=${DST_CONTEXT}" >> settings.local.sh
+
+  # printStatusMsg "Is your destination cluster on OpenShift Platform 4? (yes/no):"
+  # read DST_OCP4_RESP
+  # if [ "${DST_OCP4_RESP}" = "yes" ]; then 
+  #   DST_OCP4=true 
+  # else 
+  #   DST_OCP4=false 
+  # fi
+  # echo "export DST_OCP4=${DST_OCP4}" >> settings.sh
+
+  printStatusMsg "Please enter your source cluster from the list above:"
+  read SRC_CONTEXT
+  echo "export SRC_CONTEXT=${SRC_CONTEXT}" >> settings.local.sh
+
+  # printStatusMsg "Is your source cluster on OpenShift Platform 4? (yes/no):"
+  # read SRC_OCP4_RESP
+  # if [ "${SRC_OCP4_RESP}" = "yes" ]; then 
+  #   SRC_OCP4=true 
+  # else 
+  #   SRC_OCP4=false 
+  # fi
+  # echo "export SRC_OCP4=${SRC_OCP4}" >> settings.sh
+
+  printStatusMsg "Please enter your source PVC"
+  read SRC_PVC
+  echo "export SRC_PVC=${SRC_PVC}" >> settings.local.sh
+
+  printStatusMsg "Please enter your destination PVC"
+  read DST_PVC
+  echo "export DST_PVC=${DST_PVC}" >> settings.local.sh
+
+  REMOTE_CLUSTER=$(echo ${DST_CONTEXT} | awk 'BEGIN { FS = "/" } ; { print $2 }' | sed 's/-/./g' )
+  DST_CLUSTER=$(echo ${DST_CONTEXT} | awk 'BEGIN { FS = "/" } ; { print $1 }' )
+  REMOVE_PROJECT=${DST_CLUSTER}
+  SRC_CLUSTER=$(echo ${SRC_CONTEXT} | awk 'BEGIN { FS = "/" } ; { print $1 }' )
+  DEPLOYMENT_ENV_NAME=$(echo ${SRC_CLUSTER} | sed 's/.*-//g')
+  echo "
+export REMOTE_CLUSTER=${REMOTE_CLUSTER}
+export REMOTE_PROJECT=${REMOVE_PROJECT}
+export DST_CLUSTER=${DST_CLUSTER}
+export SRC_CLUSTER=${SRC_CLUSTER}
+export DEPLOYMENT_ENV_NAME=${DEPLOYMENT_ENV_NAME}
+  " >> settings.local.sh
+
+  echo "Generating param files ..."
+  genTemplateParams.sh -lf -p default
+
+  # rename files to correct profile
+  SRC_PARAM_NAME="./templates/source-pvc-migrator/source-pvc-migrator-deploy.local.param"
+  DST_PARAM_NAME="./templates/target-pvc-migrator/target-pvc-migrator-deploy.local.param"
+
+  
+
+  updateMigratorDeplymentParams ${SRC_PVC} ${DST_PVC} ${REMOTE_CLUSTER} ${REMOTE_PROJECT} ${SRC_CLUSTER} ${DST_CLUSTER} "${SRC_PARAM_NAME} ${DST_PARAM_NAME}"
```

Comment:
Nit.: typo, should be `updateMigratorDeploymentParams`

Reply:
change completed